extractor/src/main/java/org/schabi/newpipe/extractor/services/media_ccc/linkHandler/MediaCCCConferenceLinkHandlerFactory.java

```diff
@@ -14,19 +18,43 @@ public String getUrl(String id, List<String> contentFilter, String sortFilter) t
     }
 
     @Override
-    public String getId(String url) throws ParsingException {
-        if(url.startsWith("https://api.media.ccc.de/public/conferences/")) {
-            return url.replace("https://api.media.ccc.de/public/conferences/", "");
-        } else if(url.startsWith("https://media.ccc.de/c/")) {
-            return Parser.matchGroup1("https://media.ccc.de/c/([^?#]*)", url);
-        } else {
-            throw new ParsingException("Could not get id from url: " + url);
+    public String getId(String urlString) throws ParsingException {
+        if (urlString.startsWith("https://api.media.ccc.de/public/conferences/")) {
+            return urlString.replace("https://api.media.ccc.de/public/conferences/", "");
+        } else if (urlString.startsWith("https://media.ccc.de/c/")) {
+            return Parser.matchGroup1("https://media.ccc.de/c/([^?#]*)", urlString);
         }
+
+        URL url;
+        try {
+            url = Utils.stringToURL(urlString);
+        } catch (MalformedURLException e) {
+            throw new IllegalArgumentException("The given URL is not valid");
+        }
+
+        String path = url.getPath();
+        // remove leading "/" of URL-path if URL-path is given
+        if (!path.isEmpty()) {
+            path = path.substring(1);
+        }
+
+        if (path.contains("b/")) {
+            return path.substring(2);
+        }
+
+        throw new ParsingException("Could not get id from url: " + url);
+
     }
 
     @Override
     public boolean onAcceptUrl(String url) throws ParsingException {
-        return url.startsWith("https://api.media.ccc.de/public/conferences/")
-                || url.startsWith("https://media.ccc.de/c/");
+        try {
+            getId(url);
+            return true;
+        } catch (FoundAdException fe) {
```

Comment:
Why are you catching that FoundAdException? `getId` doesn't even throw that.

Reply:
my bad. because I used the same piece of code of youtubestreamlinkhandlerfactory and yeah, it's thrown on ad. I removed it.